dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/generate/ServiceImplementationGenerator.java

```diff
@@ -310,37 +313,35 @@ private TypeName underlyingCustomType(ArgumentType argumentType) {
 
             @Override
             public Optional<CodeBlock> customType(TypeName _typeName) {
-                if (maybeParameterEncoderType.isEmpty()) {
-                    return Optional.empty();
-                }
-
-                return maybeParameterEncoderType.flatMap(parameterEncoderType -> parameterEncoderType
-                        .type()
-                        .match(new EncoderType.Cases<Optional<CodeBlock>>() {
-                            @Override
-                            public Optional<CodeBlock> param() {
-                                return Optional.of(CodeBlock.of(
-                                        "$L.$L($S, $L.$L($L));",
-                                        REQUEST,
-                                        singleValueMethod,
-                                        key,
-                                        parameterEncoderType.encoderFieldName(),
-                                        parameterEncoderType.encoderMethodName(),
-                                        argName));
-                            }
-
-                            @Override
-                            public Optional<CodeBlock> listParam() {
-                                return Optional.of(CodeBlock.of(
-                                        "$L.$L($S, $L.$L($L));",
-                                        REQUEST,
-                                        multiValueMethod,
-                                        key,
-                                        parameterEncoderType.encoderFieldName(),
-                                        parameterEncoderType.encoderMethodName(),
-                                        argName));
-                            }
-                        }));
+                return maybeParameterEncoderType.map(this::customType);
```

Comment:
Is there a reason `generatePlainSerializer` returns an optional? The only empty return value is here and it seems like it would be better to throw if a parameter encoder is not defined for a custom type.

Reply:
Yes, so that I could build this incrementally, and ignore stuff that wasn't handled :) I think at this point the Optional could be removed.